### DISCUSSION TITLE
LDAP authentication - Charts

### DISCUSSION THREAD

**Discussion by nefelitav** [provectus/kafka-ui#3388] [Q&A] [answered]
Hi all,
I am trying to add LDAP authentication to my Kafka UI instance. In values.envs.config I added the following variables:
    **AUTH_TYPE: "LDAP"
    SPRING_LDAP_URLS: ""
    SPRING_LDAP_DN_PATTERN: ""
    SPRING_LDAP_USERFILTER_SEARCHFILTER: ""
    SPRING_LDAP_USERFILTER_SEARCHBASE: ""
    SPRING_LDAP_GROUPFILTER_SEARCHFILTER: ""
    SPRING_LDAP_GROUPFILTER_SEARCHBASE: ""
    SPRING_LDAP_ADMINUSER: ""
    SPRING_LDAP_ADMINPASSWORD: ""**

I'm not sure these variables actually exist but I couldn't find any documentation on this. Also, I get the following logs in the Kafka UI pod: 
**INFO [main] c.p.k.u.c.a.LdapSecurityConfig: Configuring LDAP authentication.
INFO [main] o.s.l.c.s.AbstractContextSource: Property 'userDn' not set - anonymous context will be used for read-write operations**

But LDAP doesn't seem to work.
Could you please tell me what the corresponding userDN variable is and in general which variables do I need to add to my values.yaml to configure LDAP?

Thanks.

**Comment by Haarolean** [comment, marked as answer]
Hey, [this](https://github.com/provectus/kafka-ui/blob/master/documentation/compose/auth-ldap.yaml) example does contain all the variables present at the moment. 

Cheers

**Reply by nefelitav** [reply]
Thank you for your answer. LDAP still doesn't work though. Should I add/change something else in my helm charts? Should I wait for the 0.6 release?
Best regards

**Reply by Haarolean** [reply]
Welp, it might be some flaky issue with LDAP filters. I've heard about it once but couldn't help without a way to reproduce the issue.
The example we have `auth-ldap.yml` does work fine, though.

Can you share the values for your searchfilters/patterns? Let's try to have a look into this.
